aea/aea_builder.py

```diff
@@ -1589,11 +1623,14 @@ def set_from_configuration(
         if agent_configuration.decision_maker_handler != {}:
             dotted_path = agent_configuration.decision_maker_handler["dotted_path"]
             file_path = agent_configuration.decision_maker_handler["file_path"]
+            config = agent_configuration.error_handler["config"]
             self.set_decision_maker_handler_paths(dotted_path, file_path)
+            self.set_decision_maker_handler_config(config)
         if agent_configuration.error_handler != {}:
             dotted_path = agent_configuration.error_handler["dotted_path"]
             file_path = agent_configuration.error_handler["file_path"]
-            self.set_error_handler(dotted_path, file_path)
+            config = agent_configuration.error_handler["config"]
+            self.set_error_handler(dotted_path, file_path, config)
```

Comment:
I wonder why we have `set_decision_maker_handler_paths` and `set_decision_maker_handler_config` but for handler we have only `set_error_handler` (maybe it is trivial, I haven't checked)

Reply:
updated

Reply:
ok

Reply:
(the decision maker approach is the correct one as it guarantees loading happens **after** the protocols, contracts and connections are loaded, so error handler can depend on them (as it should be able to))